src/integrationTest/java/com/hivemq/helmcharts/CompatibilityHelmChartDeploymentIT.java

```diff
@@ -25,7 +25,7 @@ public class CompatibilityHelmChartDeploymentIT {
 
     @Timeout(value = 20, unit = TimeUnit.MINUTES)
     @ParameterizedTest
-    @ValueSource(strings = {"v1.20.15-k3s1", "v1.21.10-k3s1", "v1.22.7-k3s1", "v1.23.4-k3s1","v1.24.3-k3s1"})
+    @ValueSource(strings = {"v1.20.15-k3s1", "v1.21.10-k3s1", "v1.22.7-k3s1", "v1.23.4-k3s1","v1.24.3-k3s1","v1.25.8-k3s1","v1.26.3-k3s1"})
```

Comment:
```suggestion
    @ValueSource(strings = {"v1.20.15-k3s1", "v1.21.14-k3s1", "v1.22.17-k3s1", "v1.23.17-k3s1", "v1.24.11-k3s1", "v1.25.8-k3s1", "v1.26.3-k3s1"})
```

Reply:
Hmm, I would suggest the following patch to put the version numbers into a central place:
```
Index: src/integrationTest/java/com/hivemq/helmcharts/CompatibilityHelmChartDeploymentIT.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/integrationTest/java/com/hivemq/helmcharts/CompatibilityHelmChartDeploymentIT.java b/src/integrationTest/java/com/hivemq/helmcharts/CompatibilityHelmChartDeploymentIT.java
--- a/src/integrationTest/java/com/hivemq/helmcharts/CompatibilityHelmChartDeploymentIT.java	(revision 22c6f8dae38ebb57f2779b7e2b41ca9e450f5e86)
+++ b/src/integrationTest/java/com/hivemq/helmcharts/CompatibilityHelmChartDeploymentIT.java	(date 1680259668138)
@@ -3,11 +3,12 @@
 import com.hivemq.client.mqtt.MqttGlobalPublishFilter;
 import com.hivemq.client.mqtt.datatypes.MqttQos;
 import com.hivemq.client.mqtt.mqtt5.Mqtt5Client;
+import com.hivemq.helmcharts.util.DockerImageNames;
 import com.hivemq.helmcharts.util.OperatorHelmChartContainer;
 import org.jetbrains.annotations.NotNull;
 import org.junit.jupiter.api.Timeout;
 import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.jupiter.params.provider.EnumSource;
 import org.testcontainers.junit.jupiter.Testcontainers;
 
 import java.nio.charset.StandardCharsets;
@@ -25,9 +26,9 @@
 
     @Timeout(value = 20, unit = TimeUnit.MINUTES)
     @ParameterizedTest
-    @ValueSource(strings = {"v1.20.15-k3s1", "v1.21.10-k3s1", "v1.22.7-k3s1", "v1.23.4-k3s1","v1.24.3-k3s1","v1.25.8-k3s1","v1.26.3-k3s1"})
-    public void withHelmLocalVersionDeployment_mqttMessagePublishedReceived(final @NotNull String version) throws Exception {
-        try (final var container = new OperatorHelmChartContainer(version, "k3s.dockerfile",
+    @EnumSource(value = DockerImageNames.K3s.class)
+    public void withHelmLocalVersionDeployment_mqttMessagePublishedReceived(final @NotNull DockerImageNames.K3s k3s) throws Exception {
+        try (final var container = new OperatorHelmChartContainer(k3s, "k3s.dockerfile",
                 "values/customTestValues.yaml")) {
             container.withLocalImages();
             container.start();
@@ -40,7 +41,6 @@
             client.connect();
 
             try (final var publishes = client.publishes(MqttGlobalPublishFilter.ALL)) {
-
                 client.subscribeWith().topicFilter("test").send();
                 client.publishWith()
                         .topic("test")
Index: src/integrationTest/java/com/hivemq/helmcharts/LocalHelmChartDeploymentIT.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/integrationTest/java/com/hivemq/helmcharts/LocalHelmChartDeploymentIT.java b/src/integrationTest/java/com/hivemq/helmcharts/LocalHelmChartDeploymentIT.java
--- a/src/integrationTest/java/com/hivemq/helmcharts/LocalHelmChartDeploymentIT.java	(revision 22c6f8dae38ebb57f2779b7e2b41ca9e450f5e86)
+++ b/src/integrationTest/java/com/hivemq/helmcharts/LocalHelmChartDeploymentIT.java	(date 1680259668130)
@@ -3,6 +3,7 @@
 import com.hivemq.client.mqtt.MqttGlobalPublishFilter;
 import com.hivemq.client.mqtt.datatypes.MqttQos;
 import com.hivemq.client.mqtt.mqtt5.Mqtt5Client;
+import com.hivemq.helmcharts.util.DockerImageNames;
 import com.hivemq.helmcharts.util.OperatorHelmChartContainer;
 import org.jetbrains.annotations.NotNull;
 import org.junit.jupiter.api.Test;
@@ -25,7 +26,7 @@
 
     @Container
     private final @NotNull OperatorHelmChartContainer
-            container = new OperatorHelmChartContainer("v1.24.3-k3s1",
+            container = new OperatorHelmChartContainer(DockerImageNames.K3s.V1_24,
             "k3s.dockerfile",
             "values/customTestValues.yaml")
             .withLocalImages();
@@ -41,7 +42,6 @@
         client.connect();
 
         try (final var publishes = client.publishes(MqttGlobalPublishFilter.ALL)) {
-
             client.subscribeWith().topicFilter("test").send();
             client.publishWith()
                     .topic("test")
Index: src/integrationTest/resources/k3s.dockerfile
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/integrationTest/resources/k3s.dockerfile b/src/integrationTest/resources/k3s.dockerfile
--- a/src/integrationTest/resources/k3s.dockerfile	(revision 22c6f8dae38ebb57f2779b7e2b41ca9e450f5e86)
+++ b/src/integrationTest/resources/k3s.dockerfile	(date 1680259718242)
@@ -1,4 +1,4 @@
-ARG K3S_VERSION=v1.21.10-k3s1
+ARG K3S_VERSION=v1.21.14-k3s1
 
 FROM ubuntu:20.04 AS builder
 RUN apt-get update && apt-get install curl gnupg2 apt-transport-https ca-certificates -yq
@@ -7,4 +7,3 @@
 
 FROM rancher/k3s:${K3S_VERSION}
 COPY --from=builder /usr/local/bin/helm /bin/
-
Index: src/integrationTest/java/com/hivemq/helmcharts/util/OperatorHelmChartContainer.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/integrationTest/java/com/hivemq/helmcharts/util/OperatorHelmChartContainer.java b/src/integrationTest/java/com/hivemq/helmcharts/util/OperatorHelmChartContainer.java
--- a/src/integrationTest/java/com/hivemq/helmcharts/util/OperatorHelmChartContainer.java	(revision 22c6f8dae38ebb57f2779b7e2b41ca9e450f5e86)
+++ b/src/integrationTest/java/com/hivemq/helmcharts/util/OperatorHelmChartContainer.java	(date 1661872645608)
@@ -37,10 +37,10 @@
     private final @NotNull List<String> imagesNames;
     private boolean withCustomImages = false;
 
-    public OperatorHelmChartContainer(final @NotNull String k3sVersion,
+    public OperatorHelmChartContainer(final @NotNull DockerImageNames.K3s k3s,
                                       final @NotNull String dockerfileName,
                                       final @NotNull String customValuesFile) {
-        super(getAdHocImageName(k3sVersion, dockerfileName));
+        super(getAdHocImageName(k3s, dockerfileName));
         super.addExposedPort(MQTT_PORT);
         super.withCopyFileToContainer(MountableFile.forHostPath("./charts/hivemq-operator"), "/chart");
         super.withCopyFileToContainer(MountableFile.forClasspathResource(customValuesFile), "/files/values.yml");
@@ -48,12 +48,12 @@
         imagesNames = new ArrayList<>();
     }
 
-    private static @NotNull DockerImageName getAdHocImageName(final @NotNull String k3sVersion,
+    private static @NotNull DockerImageName getAdHocImageName(final @NotNull DockerImageNames.K3s k3s,
                                                               final @NotNull String dockerfileName) {
         final var dockerfile = new File(MountableFile.forClasspathResource(dockerfileName).getFilesystemPath());
 
         final var s = new ImageFromDockerfile().withDockerfile(dockerfile.toPath())
-                .withBuildArg("K3S_VERSION", k3sVersion)
+                .withBuildArg("K3S_VERSION", k3s.getVersion())
                 .get();
 
         return DockerImageName.parse(s).asCompatibleSubstituteFor("rancher/k3s");
Index: src/integrationTest/java/com/hivemq/helmcharts/LocalImageDeploymentIT.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/integrationTest/java/com/hivemq/helmcharts/LocalImageDeploymentIT.java b/src/integrationTest/java/com/hivemq/helmcharts/LocalImageDeploymentIT.java
--- a/src/integrationTest/java/com/hivemq/helmcharts/LocalImageDeploymentIT.java	(revision 22c6f8dae38ebb57f2779b7e2b41ca9e450f5e86)
+++ b/src/integrationTest/java/com/hivemq/helmcharts/LocalImageDeploymentIT.java	(date 1680259668102)
@@ -1,5 +1,6 @@
 package com.hivemq.helmcharts;
 
+import com.hivemq.helmcharts.util.DockerImageNames;
 import org.jetbrains.annotations.NotNull;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.Timeout;
@@ -22,7 +23,7 @@
 
     @Container
     private final @NotNull K3sContainer container = new K3sContainer(DockerImageName
-            .parse("rancher/k3s:v1.26.3-k3s1"))
+            .parse(DockerImageNames.K3s.V1_26.getImage()))
             .withFileSystemBind("./build/containers", "/build", BindMode.READ_ONLY);
 
     @Timeout(value = 2, unit = TimeUnit.MINUTES)
@@ -38,5 +39,4 @@
         final var outListImages = container.execInContainer("/bin/ctr", "images", "ls");
         assertTrue(outListImages.getStdout().contains("hivemq/hivemq4-test"));
     }
-
 }
Index: src/integrationTest/java/com/hivemq/helmcharts/UserPermissionsIT.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/integrationTest/java/com/hivemq/helmcharts/UserPermissionsIT.java b/src/integrationTest/java/com/hivemq/helmcharts/UserPermissionsIT.java
--- a/src/integrationTest/java/com/hivemq/helmcharts/UserPermissionsIT.java	(revision 22c6f8dae38ebb57f2779b7e2b41ca9e450f5e86)
+++ b/src/integrationTest/java/com/hivemq/helmcharts/UserPermissionsIT.java	(date 1680259689322)
@@ -3,6 +3,7 @@
 import com.hivemq.client.mqtt.MqttGlobalPublishFilter;
 import com.hivemq.client.mqtt.datatypes.MqttQos;
 import com.hivemq.client.mqtt.mqtt5.Mqtt5Client;
+import com.hivemq.helmcharts.util.DockerImageNames;
 import com.hivemq.helmcharts.util.OperatorHelmChartContainer;
 import org.jetbrains.annotations.NotNull;
 import org.junit.jupiter.api.Test;
@@ -22,7 +23,7 @@
 
     @Container
     private final @NotNull OperatorHelmChartContainer
-            container = new OperatorHelmChartContainer("v1.23.4-k3s1",
+            container = new OperatorHelmChartContainer(DockerImageNames.K3s.V1_23,
             "k3s.dockerfile",
             "values/permissionsDeployment.yaml")
             .withLocalImages("hivemq-k8s-test-rootless.tar");
@@ -39,7 +40,6 @@
         client.connect();
 
         try (final var publishes = client.publishes(MqttGlobalPublishFilter.ALL)) {
-
             client.subscribeWith().topicFilter("test").send();
             client.publishWith()
                     .topic("test")
Index: src/integrationTest/java/com/hivemq/helmcharts/util/DockerImageNames.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/integrationTest/java/com/hivemq/helmcharts/util/DockerImageNames.java b/src/integrationTest/java/com/hivemq/helmcharts/util/DockerImageNames.java
new file mode 100644
--- /dev/null	(date 1680259668122)
+++ b/src/integrationTest/java/com/hivemq/helmcharts/util/DockerImageNames.java	(date 1680259668122)
@@ -0,0 +1,33 @@
+package com.hivemq.helmcharts.util;
+
+import org.jetbrains.annotations.NotNull;
+
+public class DockerImageNames {
+
+    public enum K3s {
+        V1_20("v1.20.15-k3s1"),
+        V1_21("v1.21.14-k3s1"),
+        V1_22("v1.22.17-k3s1"),
+        V1_23("v1.23.17-k3s1"),
+        V1_24("v1.24.11-k3s1"),
+        V1_25("v1.25.8-k3s1"),
+        V1_26("v1.26.3-k3s1");
+
+        private final @NotNull String version;
+
+        K3s(final @NotNull String version) {
+            this.version = version;
+        }
+
+        public @NotNull String getVersion() {
+            return version;
+        }
+
+        public @NotNull String getImage() {
+            return "rancher/k3s:" + version;
+        }
+    }
+
+    private DockerImageNames() {
+    }
+}
```